RecoTracker/TkSeedingLayers/interface/SeedingHitSet.h

```diff
@@ -11,40 +19,60 @@ class SeedingHitSet {
 
   static ConstRecHitPointer nullPtr() { return nullptr; }
 
-  SeedingHitSet() { theRecHits[0] = theRecHits[1] = theRecHits[2] = theRecHits[3] = nullptr; }
+  SeedingHitSet() { initContainer(); }
 
-  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two)
-  // : theRecHits{{one,two,ConstRecHitPointer()}}
-  {
+  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two) {
+    initContainer();
     theRecHits[0] = one;
     theRecHits[1] = two;
-    theRecHits[2] = theRecHits[3] = nullptr;
+    setSize();
   }
-  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three)
-  // : theRecHits{{one,two,three}},
-  {
+  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three) {
+    initContainer();
     theRecHits[0] = one;
     theRecHits[1] = two;
     theRecHits[2] = three;
-    theRecHits[3] = nullptr;
+    setSize();
   }
 
   SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three, ConstRecHitPointer four) {
+    initContainer();
     theRecHits[0] = one;
     theRecHits[1] = two;
     theRecHits[2] = three;
     theRecHits[3] = four;
+    setSize();
+  }
+
+  SeedingHitSet(std::vector<ConstRecHitPointer> &hits) {
+    if (hits.size() > seedinghits::kMaxSeedSize)
+      edm::LogWarning("SeedingHitSet") << "More than " << seedinghits::kMaxSeedSize
+                                       << " hits give in input. Taking only the first " << seedinghits::kMaxSeedSize;
+
+    initContainer();
+    std::copy(hits.begin(), hits.begin() + std::min(hits.size(), size_t(seedinghits::kMaxSeedSize)), theRecHits);
+    setSize();
   }
 
   ConstRecHitPointer const *data() const { return theRecHits; }
 
-  unsigned int size() const { return theRecHits[3] ? 4 : (theRecHits[2] ? 3 : (theRecHits[1] ? 2 : 0)); }
```

Comment:
@slava77 `theSize` is set to `0` for one hit case to mimic the behavior here where 0th hit is not even checked. Intuitively, I think that the logic here was: if a SeedingHitSet has only one valid hit it's not a proper set. Then I don't know how much this is actually used. There are some places where changing this would change the behavior (see e.g. [here](https://cmssdt.cern.ch/dxr/CMSSW/source/RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc#97)). But most probably these occurrences are protected in other ways before. Nevertheless I would leave it as it is.